tools/scenario-player/scenario_player/utils.py

```diff
@@ -187,23 +186,25 @@ def wait_for_txs(client_or_web3, txhashes, timeout=360):
         )
 
 
-def get_or_deploy_token(runner) -> ContractProxy:
+def get_or_deploy_token(runner) -> Tuple[ContractProxy, int]:
     """ Deploy or reuse  """
-    contract_manager = ContractManager(contracts_precompiled_path())
-    token_contract = contract_manager.get_contract(CONTRACT_CUSTOM_TOKEN)
+    token_contract = runner.contract_manager.get_contract(CONTRACT_CUSTOM_TOKEN)
 
     token_config = runner.scenario.get('token', {})
     if not token_config:
         token_config = {}
     address = token_config.get('address')
+    block = token_config.get('block', 0)
     reuse = token_config.get('reuse', False)
 
     token_address_file = runner.data_path.joinpath('token.addr')
     if reuse:
         if address:
             raise ScenarioError('Token settings "address" and "reuse" are mutually exclusive.')
         if token_address_file.exists():
-            address = token_address_file.read_text()
+            token_data = json.loads(token_address_file.read_text())
```

Comment:
This will raise a `JSONDecodeError` for existing scenarios that already saved an address. Using another filename would avoid that problem.